frontend/src/components/student/AllActivities/CombatTask/CombatTask.js

```diff
@@ -81,69 +78,131 @@ export default function CombatTask() {
-              <h5>{task.description}</h5>
-              {task.points !== null && (
+  //add overflows where needed
+
+  const Header = () => (
+    <>
+      <ActivityImg src={getActivityImg('TASK')}></ActivityImg>
+      <ActivityType>{getActivityTypeName('TASK')}</ActivityType>
+      <ActivityName>{task.name}</ActivityName>
+    </>
+  )
+
+  const VerticalSpacer = () => <Row style={{ height: '2vh' }}></Row>
+
+  const HorizontalSpacer = () => <Col style={{ height: '3vh' }} />
+
+  const ActivityDetails = () => (
+    <Col>
+      <h2>Treść:</h2>
+      <p>{task.description}</p>
+    </Col>
+  )
+
+  const AwaitingFeedbackField = () => (
+    <Col md={6} className={'text-center p-4 my-auto'}>
+      <FontAwesomeIcon className={'m-2'} icon={faHourglass} size='5x' spin />
+      <h2 className={'m-2'}>Odpowiedź została przesłana, oczekiwanie na sprawdzenie przez prowadzącego</h2>
+    </Col>
+  )
+
+  const content = () => {
+    return !task && errorMessage === '' ? (
+      <Loader />
+    ) : errorMessage !== '' ? (
+      <p>{errorMessage}</p>
+    ) : (
+      <>
+        <HorizontalSpacer />
+        <Col
+          className='m-0 pt-4 mx-auto'
+          style={{ height: '94vh', width: '90%', backgroundColor: 'var(--light-blue)' }}
+        >
+          <HeaderRow className='p-2 rounded mx-2' style={{ backgroundColor: 'var(--dark-blue)', height: '8vh' }}>
+            <Header />
+          </HeaderRow>
+          <VerticalSpacer />
+          <Row
+            className='p-2 rounded mx-2 overflow-auto'
+            style={{ backgroundColor: 'var(--dark-blue)', height: '25vh' }}
+          >
+            <ActivityDetails />
+          </Row>
+          <VerticalSpacer />
+          <Row className='p-2 rounded mx-2' style={{ backgroundColor: 'var(--dark-blue)', height: '50vh' }}>
+            <Col
+              md={task.answer || hasSentNow ? MD_WHEN_TASK_SENT : MD_WHEN_TASK_NOT_SENT}
+              className={'h-100 overflow-auto'}
+            >
+              <h4>Odpowiedź:</h4>
+              {isRevieved() ? (
+                <Col>
+                  <h4>Twoja odpowiedź</h4>
+                  <p>{task.answer}</p>
+                </Col>
+              ) : (
                 <>
-                  <SmallDivider />
-                  <p>
-                    Zdobyte punkty: <strong>{task.points}</strong>
-                  </p>
-                  <p>Uwagi od prowadzącego:</p>
-                  <p>{task.remarks ?? 'Brak uwag'}</p>
-                  {task.feedbackFile && <FeedbackFileService feedbackFile={task.feedbackFile} />}
-                </>
-              )}
-              <SmallDivider />
-              {task.points == null && (
-                <RemarksCol>
-                  <h4>Odpowiedź:</h4>
+                  {task.answer ? (
+                    <Col>
+                      <h5>Twoja obecna odpowiedź</h5>
+                      <p>{task.answer}</p>
+                    </Col>
+                  ) : null}
```

Comment:
A takie zapisanie nie będzie lepsze? I czy zadziała bez zmian?:
```jsx
{task.answer && (
  <Col>
    <h5>Twoja obecna odpowiedź</h5>
    <p>{task.answer}</p>
  </Col>
)}
```
Jeżeli z jakiegoś powodu nie można zrobić jw to zamienił bym na `<></>` na końcu, bo ten `null` wygląda mega dziwnie